hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -127,24 +130,25 @@ public static Schema toSchema(HCatSchema hiveSchema) {
   private static Schema.Type getType(String name, PrimitiveObjectInspector.PrimitiveCategory category) {
     switch (category) {
       case BOOLEAN:
-        return Schema.Type.BOOLEAN;
+        return Schema.nullableOf(Schema.of(Schema.Type.BOOLEAN)).getType();
       case BYTE:
       case CHAR:
       case SHORT:
       case INT:
-        return Schema.Type.INT;
+        return Schema.nullableOf(Schema.of(Schema.Type.INT)).getType();
```

Comment:
do byte, char, and short fields work we convert a HCatRecord into a StructuredRecord?